File: python/两数之和.py

from typing import List 

class Solution:
    def twoSum(self , numbers: List[int], target: int) -> List[int]:
        res = []
        #创建哈希表,两元组分别表示值、下标
        hash = dict() 
        #在哈希表中查找target-numbers[i]
        for i in range(len(numbers)) :
            temp = target - numbers[i]
            #若是没找到，将此信息计入哈希表
            if temp not in hash: 
                hash[numbers[i]] = i
            else:
                #哈希表中记录的是之前的数字，所以该索引比当前小
                res.append(hash[temp] + 1) 
                res.append(i + 1)
                break
        return res

File: C/两数之和.cpp
#include<iostream>
#include<cmath>
#include<string>
#include<vector>
#include <unordered_map>

using namespace std;

class Solution 
{
public:
    /**
     *
     * @param numbers int整型vector
     * @param target int整型
     * @return int整型vector
     */
    vector<int> twoSum(vector<int>& numbers, int target)
    {
        // write code here
        int n = numbers.size();
        vector <int> res;

        if (!n)
        {
            return res;
        }

        unordered_map<int, int> hashmap;
        for (int i = 0; i < n; i++)
        {
            if (hashmap.find(target - numbers[i]) != hashmap.end())
            {
                res.push_back(hashmap[target - numbers[i]] + 1);
                res.push_back(i + 1);
            }
            else
            {
                hashmap[numbers[i]] = i;
            }
        }
        return res;
    }
};